infer/src/bo/bufferOverrunChecker.ml

```diff
@@ -0,0 +1,470 @@
+      (cond_set, mem)
+
+    let collect_instrs
+      : extras ProcData.t -> CFG.node -> Sil.instr list -> Dom.Mem.t
+        -> Dom.ConditionSet.t -> Dom.ConditionSet.t
+    = fun pdata node instrs mem cond_set ->
+      IList.fold_left (collect_instr pdata node) (cond_set, mem) instrs
+      |> fst
+
+    let collect_node
+      : extras ProcData.t -> Analyzer.invariant_map -> Dom.ConditionSet.t ->
+        CFG.node -> Dom.ConditionSet.t
+    = fun pdata inv_map cond_set node ->
+      let instrs = CFG.instr_ids node |> IList.map fst in
+      match Analyzer.extract_pre (CFG.id node) inv_map with
+      | Some mem -> collect_instrs pdata node instrs mem cond_set
+      | _ -> cond_set
+
+    let collect
+      : extras ProcData.t -> Analyzer.invariant_map -> Dom.ConditionSet.t
+    = fun ({ pdesc } as pdata) inv_map ->
+      let add_node1 acc node = collect_node pdata inv_map acc node in
+      Procdesc.fold_nodes add_node1 Dom.ConditionSet.empty pdesc
+
+    let report_error : Tenv.t -> Procdesc.t -> Dom.ConditionSet.t -> unit
+    = fun tenv pdesc conds ->
+      let pname = Procdesc.get_proc_name pdesc in
+      let report1 cond =
+        let safe = Dom.Condition.check cond in
+        let (caller_pname, loc) =
+          match Dom.Condition.get_trace cond with
+          | Dom.Condition.Inter (caller_pname, _, loc) -> (caller_pname, loc)
+          | Dom.Condition.Intra pname -> (pname, Dom.Condition.get_location cond)
+        in
+        (* report symbol-related alarms only in debug mode *)
+        if not safe && Procname.equal pname caller_pname then
+          let bo_kind = "BUFFER-OVERRUN CHECKER" in
+          let cond_str = Dom.Condition.to_string cond in
+          Checkers.ST.report_error tenv pname pdesc bo_kind loc cond_str
+      in
+      Dom.ConditionSet.iter report1 conds
+  end
+
+  let analyze_ondemand_
+    : Tenv.t -> Analyzer.TransferFunctions.extras -> Procdesc.t
+      -> Dom.Summary.t option
+  = fun tenv get_pdesc pdesc ->
+    let cfg = CFG.from_pdesc pdesc in
+    let pdata = ProcData.make pdesc tenv get_pdesc in
+    let pname = Procdesc.get_proc_name pdesc in
+    let inv_map = Analyzer.exec_pdesc pdata in
+    let entry_mem =
+      let entry_id = CFG.id (CFG.start_node cfg) in
+      Analyzer.extract_post entry_id inv_map
+    in
+    let exit_mem =
+      let exit_id = CFG.id (CFG.exit_node cfg) in
+      Analyzer.extract_post exit_id inv_map
+    in
+    let cond_set = Report.collect pdata inv_map in
+    Report.report_error tenv pdesc cond_set;
+    if Procname.get_method pname = "infer_print" then None else
+      match entry_mem, exit_mem with
+      | Some entry_mem, Some exit_mem ->
+          Summary.write_summary pname (entry_mem, exit_mem, cond_set);
+          Some (entry_mem, exit_mem, cond_set)
+      | _ -> None
+
+  let checker : Callbacks.proc_callback_args -> Dom.Summary.t option
+  = fun { get_proc_desc; proc_desc; proc_name; tenv } ->
+    let analyze_ondemand _ pdesc =
```

Comment:
We recently made the `AbstractInterpreter` and `Interprocedural` module much more expressive so it should no longer be necessary to manually set up the on-demand callbacks like this. You should be able to add all of the postprocessing steps like reporting in a custom `compute_post` function that you pass via the `~compute_post` parameter to `Interprocedural`. See siof.ml and taintAnalysis.ml for examples of this.